[Observability] Add docs on schema selection for host data

mdbirnstiehl opened the pull request:
This PR closes #2084

mdbirnstiehl commented:
@roshan-elastic I've started a draft here for the schema selector docs so we can iterate on what we need. Previews are in the comment above titled **Preview links for changed docs**. 

Let me know what you think about the updates, and if we need to change anything or add additional information anywhere.

I still need to update the metrics reference page, is there a PR or documentation I can look at that has the metric definitions that need to be added?

roshan-elastic commented on solutions/observability/infra-and-hosts/detect-metric-anomalies.md:
```diff
@@ -12,17 +12,16 @@ products:
+To create ML jobs to detect metric anomalies, you need to meet the following requirements:
+
+* **For Observability serverless projects**, the **Editor** role or higher is required to create {{ml}} jobs. To learn more, refer to [Assign user roles and privileges](/deploy-manage/users-roles/cloud-organization/user-roles.md#general-assign-user-roles).
+* Metric anomaly detection does not work for OpenTelemetry hosts.
```
This looks good @mdbirnstiehl

roshan-elastic commented:
**Hosts in Infra Inventory**
One other thing @mdbirnstiehl, the Infra Inventory will show the schema selector for hosts too so we'll need to update the infra inventory docs to reflect this too:

- [ ] https://github.com/elastic/kibana/pull/230528

FYI it uses the same schema selector and behaves in the same way as hosts.

We'll be adding a link from this selector to the docs to explain how it works (if we wanted to simplify the docs we could possibly just refer to the [schema selector in the host docs](https://ela.st/schema-selector-hosts) from the inventory docs for more info?)

FYI - there's a possibility we remove schema selection from the infra inventory in a follow up issue but we haven't decided yet. Any work we do in these infra inventory docs should probably be minimal.

mdbirnstiehl commented:
@roshan-elastic I've added the following:

- Added notes to the [Processes](https://github.com/elastic/docs-content/pull/2707/files#diff-6de2fc48955b83c25ece5e34e4ed1fadda0d9372f1e2a15b26db65f3fb580f68R223) and [Anomalies](https://github.com/elastic/docs-content/pull/2707/files#diff-6de2fc48955b83c25ece5e34e4ed1fadda0d9372f1e2a15b26db65f3fb580f68R295) sections under view hosts.
- Added a [list](https://github.com/elastic/docs-content/pull/2707/files#diff-6de2fc48955b83c25ece5e34e4ed1fadda0d9372f1e2a15b26db65f3fb580f68R167) noting the main differences when selecting otel schema under view hosts. I just added the main ones, but we can add more if you think they add value.
- Added a [note ](https://github.com/elastic/docs-content/pull/2707/files#diff-3345c77751f69d699d44de16fe0a3c33146ded84c0723aa90612ea999fa83a2eR39) to the Inventory conditions section about schema dropdown. Is the paragraph before this enough generic information about an inventory rule, or do you think we should add more?
- Added [schema note](https://github.com/elastic/docs-content/pull/2707/files#diff-ae593be1748ba41a99d60f48fa4fb3bb0e6d63928473e1408c7160ee9f71e5e7R66) to the Infrastructure Inventory docs
- Created a [section](https://github.com/elastic/docs-content/pull/2707/files#diff-131bbc63876bf6ab18252e7823048d4b45762d7f3141909b58d6d30e27996a3eR20) in the host metrics reference. Haven't added the actual formulas yet. Is the google doc up to date or does it still need to be?

roshan-elastic commented:
Hey @mdbirnstiehl 

> Added notes to the [Processes](https://github.com/elastic/docs-content/pull/2707/files#diff-6de2fc48955b83c25ece5e34e4ed1fadda0d9372f1e2a15b26db65f3fb580f68R223) and [Anomalies](https://github.com/elastic/docs-content/pull/2707/files#diff-6de2fc48955b83c25ece5e34e4ed1fadda0d9372f1e2a15b26db65f3fb580f68R295) sections under view hosts.

Looks perfect for Hosts. I think we just need to update the [inventory one](https://docs-v3-preview.elastic.dev/elastic/docs-content/pull/2707/solutions/observability/infra-and-hosts/view-infrastructure-metrics-by-resource-type) too but it might be worth just removing that and linking through to the same content in the hosts (your call).

> Added a [list](https://github.com/elastic/docs-content/pull/2707/files#diff-6de2fc48955b83c25ece5e34e4ed1fadda0d9372f1e2a15b26db65f3fb580f68R167) noting the main differences when selecting otel schema under view hosts. I just added the main ones, but we can add more if you think they add value.

Looks good!

> Added a [note ](https://github.com/elastic/docs-content/pull/2707/files#diff-3345c77751f69d699d44de16fe0a3c33146ded84c0723aa90612ea999fa83a2eR39)to the Inventory conditions section about schema dropdown. Is the paragraph before this enough generic information about an inventory rule, or do you think we should add more?

Yeah, that looks good.

> Added [schema note](https://github.com/elastic/docs-content/pull/2707/files#diff-ae593be1748ba41a99d60f48fa4fb3bb0e6d63928473e1408c7160ee9f71e5e7R66) to the Infrastructure Inventory docs

Looks good!

> Created a [section](https://github.com/elastic/docs-content/pull/2707/files#diff-131bbc63876bf6ab18252e7823048d4b45762d7f3141909b58d6d30e27996a3eR20) in the host metrics reference. Haven't added the actual formulas yet. Is the google doc up to date or does it still need to be?


Yeah I think something like should work. I think OpenTelemetry should probably be a higher-level heading as there will be all the same kinds of metrics underneath it (e.g. CPU, disk, memory etc).

I'll get the doc updated now so you can populate this and re-jig as necessary?

mdbirnstiehl commented on reference/data-analysis/observability/observability-host-metrics.md:
```diff
@@ -84,8 +91,7 @@ Learn about key host metrics displayed in the Infrastructure UI:
 
-
-## Legacy metrics [legacy-metrics]
+### Legacy metrics [legacy-metrics]
```
@roshan-elastic I'm assuming the legacy metrics belong with the system integration metrics, but just wanted to make sure they wouldn't apply to OTel metrics.

roshan-elastic replied:
That's right @mdbirnstiehl !

mdbirnstiehl commented:
@roshan-elastic I've update the host metrics page, and added the notes to the Infrastructure Inventory regard anomalies and processes. One last thing to verify, the OTel formulas are only relevant for stack from 9.2 forward, correct?

roshan-elastic commented:
> @roshan-elastic I've update the host metrics page, and added the notes to the Infrastructure Inventory regard anomalies and processes. 

Awesome @mdbirnstiehl.

> One last thing to verify, the OTel formulas are only relevant for stack from 9.2 forward, correct?
 
That will be the next major release they go into yes. They will likely go out in serverless before then though - I can let you know when it's merged into `main` to give you a better idea of exactly when it'll be available?

nastasha-solomon commented on reference/data-analysis/observability/observability-host-metrics.md:
```diff
@@ -96,3 +102,75 @@ The UI and any new rules you create will use the new metric definition. However,
+
+| Metric | Description |
+| --- | --- |
+| **CPU Usage (%)** | Average of percentage of CPU time spent in states other than Idle and IOWait, normalized by the number of CPU cores. Includes both time spent on user space and kernel space. 100% means all CPUs of the host are busy.<br><br>**Field Calculation**: `1-(average(metrics.system.cpu.utilization,kql='state: idle') + average(metrics.system.cpu.utilization,kql='state: wait'))`<br> |
```
```suggestion
| **CPU Usage (%)** | Average percentage of CPU time spent in states other than Idle and IOWait, normalized by the number of CPU cores. Includes both time spent on user space and kernel space. 100% means all CPUs of the host are busy.<br><br>**Field Calculation**: `1-(average(metrics.system.cpu.utilization,kql='state: idle') + average(metrics.system.cpu.utilization,kql='state: wait'))`<br> |
```

nastasha-solomon commented on solutions/observability/infra-and-hosts/analyze-compare-hosts.md:
```diff
@@ -36,9 +36,12 @@ To learn more about the metrics shown on this page, refer to the [Metrics refere
-If you haven’t added data yet, click **Add data** to search for and install an Elastic integration.
 
-Need help getting started? Follow the steps in [Get started with system metrics](/solutions/observability/infra-and-hosts/get-started-with-system-metrics.md).
+If you haven’t added data yet, click **Add data → Host** and select how you want to monitor your host—OpenTelemetry or Elastic Agent.
```
Not sure if you need to add the `{applies_to}` tag here. Flagging in case you do. 


```suggestion

`{applies_to}`{stack: "ga 9.2", serverless: "ga"}`

If you haven’t added data yet, click **Add data → Host** and select how you want to monitor your host—OpenTelemetry or Elastic Agent.
```

mdbirnstiehl replied:
The **Add data** steps and OTel option did exist in 9.0, but this integration language was out of date. OTel data has been available, it's just the schema selection that is new.

nastasha-solomon commented on solutions/observability/infra-and-hosts/view-infrastructure-metrics-by-resource-type.md:
```diff
@@ -57,19 +57,23 @@ You can also use the search bar to create structured queries using [{{kib}} Quer
 
+::::{note}
+{applies_to}`{stack: "ga 9.2", serverless: "ga"}`
+When showing `Hosts`, the **Schema** dropdown menu shows the available data collection schemas for the current query. If data from both the Elastic System integration and OpenTelemetry is available, the schema defaults to **OpenTelemetry**. Select **Elastic System Integration** to see data collected by the Elastic System integration.
```
Making this suggestion if the page or the content on it is literally labeled hosts. 

```suggestion
When showing **Hosts**, the **Schema** dropdown menu shows the available data collection schemas for the current query. If data from both the Elastic System integration and OpenTelemetry is available, the schema defaults to **OpenTelemetry**. Select **Elastic System Integration** to see data collected by the Elastic System integration.
```

nastasha-solomon reviewed (approved):
Just a few minor comments. Otherwise, lgtm!
